server/src/main/java/io/crate/types/NumericType.java

```diff
@@ -200,6 +200,11 @@ public TypeSignature getTypeSignature() {
         }
     }
 
+    @Override
+    public Integer characterMaximumLength() {
```

Comment:
This feels like a bit of a hack to be honest, but it fixes things for now.  In a follow-up we should change the TypeCompatibility code to delegate things into the types themselves rather than having some special-case logic for types that implement this values.

Reply:
++, I can do that.  I'll open a separate PR for it I think, as it's its own useful bit of work.

It does mean that `TypeCompatibility#typeCompatibilityForParametrizedType` will now have three different special cases in it, which indicates to me that we should be pushing these comparisons into the data types themselves.  But that can be done later, it's a bigger change and I don't want it to slow this bugfix down.

Reply:
`numeric` types are never actually stored, they're only generated from calculations on other rows as eg aggregations.  So would they even show up in `information_schema.columns`?

Reply:
Good point - they wouldn't show up - but still feels a bit error prone to use the wrong property.
Is it much effort to add the new one?

Unrelated to this PR - I had been wondering if we lack some kind of type family concept. So functions could bind to "numbers" or "characterSequence" and they'd work for all kinds of numbers or for char/varchar/string

Reply:
I think it's still worth adding the new property, it just makes it a bit tricky to test in isolation.